satellite/marketing/marketingweb/server.go

```diff
@@ -128,20 +170,73 @@ func (s *Server) parseTemplates() (err error) {
 	return nil
 }
 
+// timeConverter formats form time input as time.Time.
+func timeConverter(value string) reflect.Value {
+	v, err := time.Parse("2006-01-02", value)
+	if err != nil {
+		return reflect.Value{}
+	}
+	return reflect.ValueOf(v)
+}
+
+// formToStruct decodes POST form data into a new offer.
+func formToStruct(w http.ResponseWriter, req *http.Request) (o marketing.NewOffer, e error) {
+	err := req.ParseForm()
+	if err != nil {
+		return o, err
+	}
+
+	if err := decoder.Decode(&o, req.PostForm); err != nil {
+		return o, err
+	}
+	return o, nil
+}
+
+// CreateOffer handles requests to create new offers.
+func (s *Server) CreateOffer(w http.ResponseWriter, req *http.Request) {
+	o, err := formToStruct(w, req)
+	if err != nil {
+		s.log.Error("failed to convert form to struct", zap.Error(err))
+		s.serveInternalError(w, req, err)
+		return
+	}
+
+	o.Status = marketing.Active
+	reqType := mux.Vars(req)["offer_type"]
+
+	if reqType == "referral-offer" {
+		o.Type = marketing.Referral
+	} else {
+		o.Type = marketing.FreeCredit
+	}
+
+	if _, err := s.service.InsertNewOffer(context.Background(), &o); err != nil {
```

Comment:
`req.Context()`?

Reply:
I don't mind changing it but I would like to understand why I would use one over the other ?

Reply:
https://github.com/storj/storj/wiki/Style#contexts